poly-commitment/benches/ipa.rs

```diff
@@ -71,6 +71,52 @@ fn benchmark_msm(c: &mut Criterion) {
     }
 }
 
+fn benchmark_msm_parallel(c: &mut Criterion) {
+    use ark_ec::{AffineRepr, VariableBaseMSM};
+    use rayon::prelude::*;
+
+    let mut group = c.benchmark_group("MSM");
+    let mut rng = o1_utils::tests::make_test_rng(None);
+
+    let srs = SRS::<Vesta>::create(1 << 16);
+    srs.get_lagrange_basis_from_domain_size(1 << 16);
+
+    for msm_size_log in [8, 10, 12, 14, 16].into_iter() {
```

Comment:
As for ae1034f6f5a5b84ab302becc4da8613d72a488dc, I would focus on sizes we care, i.e. 16.